api/lib/infrastructure/events/EventDispatcher.js

```diff
@@ -0,0 +1,30 @@
+class EventDispatcher {
+  constructor() {
+    this._subscriptions = [];
+  }
+
+  subscribe(event, eventHandler) {
+    this._preventDuplicateSubscription(event, eventHandler);
+    this._subscriptions.push([event.prototype.constructor, eventHandler]);
```

Comment:
pourquoi utiliser un tableau ? Je pense que ce serait plus clair avec un objet.
```suggestion
    this._subscriptions.push({ event: event.prototype.constructor, eventHandler });
```